### PR TITLE
feat: add NextLink component

### PR DESCRIPTION
**Pull request by fsdiogo**
Adds the NextLink component.

### CODE REVIEW

File: src/NextLink.css [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.link {
```
Comment (paulobmarcos) [inline src/NextLink.css]: This class `link` has a great probability of creating collisions with the css of someones project.

Me and @PedroMiguelSS are trying to figure a good way to avoid this. If you guys have any suggestions, please share 🙏

**Review by paulobmarcos — CHANGES_REQUESTED** [review]

File: src/NextLink.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React, { forwardRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Link from 'next/link';
+import './NextLink.css';
```
Comment (paulobmarcos) [inline src/NextLink.js]: When we run our build script, css files won't be added to the 'es' and 'lib' folders.

One solution, and from what we have seen might be the "best" one, is to have a `dist` folder where all css would be placed and then each developer would import css files in the way that best suits their project needs.

**Review by paulobmarcos — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -1,15 +1,118 @@
...
+
+A className to apply to the component wrapper.
+
+#### newTab
+
+Type: `bool` | Required: `false` | Default: `false`
+
+If set to true, opens the link in a new tab.
+
+#### external
```
Comment (zebateira) [inline README.md:78]: can't we make this automatic by parsing `href`? or is there a situation when we need to override it?

File: src/NextLink.css (hunk 1 above)
Comment (zebateira) [reply src/NextLink.css]: a few suggestions:

- `.moxy-link`
- `.moxy-next-link`
- `.m-link`
- `.m-next-link`

**Review by AfonsoVReis — APPROVED** [review]

File: src/NextLink.js (hunk 2 above)
Comment (zebateira) [reply src/NextLink.js]: yeah, I vouch for that as well and it reinforces the fact that the styles should not be mandatory.

File: src/NextLink.css (hunk 1 above)
Comment (paulobmarcos) [reply src/NextLink.css]: I agree with you @zebateira. I would like to add one option. 

Every class or id could have a prefix. The prefix would be the initials of the package. For example:

(next-link)
- `nl_link`
- `nl_container`

This is very similar to what we have in react-carousel (`rc`)

Comment (fsdiogo) [reply src/NextLink.css]: I like `moxy-next-link`, although verbose it diminishes the chance of collision, while `nl_link` is still a bit more broad.

I'll go with the `moxy-next-link` for now 👍

Comment (paulobmarcos) [reply src/NextLink.css]: After talking with @PedroMiguelSS the above solution might not be great, since we may have collisions with other repos. For example:
- next-link (nl_container)
- next-label (nl_container)

So, I would actually be more verbose just to make sure we avoid this as much as possible :+1:

File: src/NextLink.js (hunk 2 above)
Comment (fsdiogo) [reply src/NextLink.js]: Done 👍

File: README.md (hunk 3 above)
Comment (fsdiogo) [reply README.md:78]: We probably could, but I'd argue it is best to play it safe and leave that to the responsibility of the developer. Having some kind of regex to parse that could bring us problems in the future. Do you agree?

Comment (zebateira) [reply README.md:78]: > We probably could, but I'd argue it is best to play it safe and leave that to the responsibility of the developer

We can still have it as an option for the user to override it.

> Having some kind of regex to parse that could bring us problems in the future. Do you agree?

What kind of problems do you mean? Edge cases we are not covering?

File: dist/styles.css [hunk 4]
```diff
@@ -0,0 +1,5 @@
+.moxy-next-link {
+    color: unset; /* stylelint-disable-line */
```
Comment (nlfonseca) [inline dist/styles.css]: do you test the color of `.moxy-next-link:hover`?

File: src/NextLink.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            { children }
+        </a>
+    );
+
+    if (external) {
+        return Anchor;
+    }
+
+    return (
+        <Link href={ href } prefetch={ prefetch && undefined } as={ as } scroll={ scroll } passHref>
```
Comment (nlfonseca) [inline src/NextLink.js:23]: what is `passHref`?

Comment (fsdiogo) [reply src/NextLink.js:23]: It's [a prop from next/link](https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-custom-component-that-wraps-an-a-tag):

> If the child of `Link` is a custom component that wraps an `<a>` tag, you must add `passHref` to `Link`. Without this, the `<a>` tag will not have the `href` attribute, which might hurt your site’s SEO.

File: dist/styles.css (hunk 4 above)
Comment (fsdiogo) [reply dist/styles.css]: Nope.

File: README.md (hunk 3 above)
Comment (fsdiogo) [reply README.md:78]: I can't point one specifically, that's the thing. They may not even be edge cases but still be out of our coverage. I'd be in favour of keeping this like it is.

Are there more opinions?

File: dist/styles.css (hunk 4 above)
Comment (nlfonseca) [reply dist/styles.css]: I suggest changing this for `inherit`, since is the same thing that `unset` and have more support.
https://caniuse.com/#search=inherit

File: dist/styles.css [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+    text-decoration: none;
+    cursor: pointer;
```
Comment (nlfonseca) [inline dist/styles.css]: the link already have this by default, so we don't need this.

File: dist/styles.css [hunk 7]
```diff
@@ -0,0 +1,5 @@
+.moxy-next-link {
+    color: unset; /* stylelint-disable-line */
+    text-decoration: none;
```
Comment (nlfonseca) [inline dist/styles.css]: by default this makes more sense having text-decoration, mainly because link in body text can have some visual feedback like an underline.

https://codepen.io/nlfonseca/pen/GRpZKoG

File: dist/styles.css (hunk 4 above)
Comment (fsdiogo) [reply dist/styles.css]: Agree, will do 👍

File: dist/styles.css (hunk 7 above)
Comment (fsdiogo) [reply dist/styles.css]: Agreed.

Having said that, the CSS will look like this:

```css
.moxy-next-link {
    color: inherit;
}
```

Do you guys think it still makes sense to include this CSS file? As this will almost definitely be styled by the developer, I think we should skip it.

Comment (nlfonseca) [reply dist/styles.css]: Agree!

Comment (PedroMiguelSS) [reply dist/styles.css]: I agree with @nlfonseca , we shouldn't be removing the `text-decoration`. I also agree with @fsdiogo: imo this file is unnecessary.

Comment (fsdiogo) [reply dist/styles.css]: Cool, I'll remove the CSS file then.

**Review by jdiogopeixoto — APPROVED** [review]

File: README.md [hunk 8]
```diff
@@ -1,15 +1,118 @@
...
+$ npm install @moxy/next-link
+```
+
+This library is written in modern JavaScript and is published in both CommonJS and ES module transpiled variants. If you target older browsers please make sure to transpile accordingly.
+
+## Motivation
+
+When using Next.js, we have [next/link](https://nextjs.org/docs/api-reference/next/link) to navigate between client-side routes, but when we need to navigate to external URLs an anchor tag has to be used.
+
+This component supports both, using `NextLink` we can pass an internal or external URL, using the same component for both. With the `external` prop we tell the component which type of route is and it takes care of the rest for us, rendering the `next/link` or an `<a>` tag, maintaining the same styles between the two.
```
Comment (ruimonteiro93) [inline README.md]: Just the tiniest nitpick 😅 

`(...) With the external prop we tell the component which type of route is and it takes care of the rest for us (...)`

I would slightly rephrase it to 

`(...) With the external prop we specify the type of route and it takes care of the rest for us (...)`

Just because `which type of route is` sounds a bit off. What do you think?

**Review by paulobmarcos — APPROVED** [review]

**Review by PedroMiguelSS — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (paulobmarcos) [reply README.md:78]: What would be our external condition? If it is a uri?

File: README.md (hunk 8 above)
Comment (fsdiogo) [reply README.md]: I like it! Fixed.

**Review by nlfonseca — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (fsdiogo) [reply README.md:78]: We'll merge this as is, but [an issue](https://github.com/moxystudio/next-link/issues/2) has been opened for further discussion.

Comment (zebateira) [reply README.md:78]: sounds good @fsdiogo !
